alf/algorithms/ppg/disjoint_policy_value_network.py

```diff
@@ -0,0 +1,244 @@
+            output action. The type of each invidual projection network in the
+            output is derived from this.
+
+    """
+
+    def _create_individually(spec):
+        constructor = (discrete_projection_net_ctor
+                       if spec.is_discrete else continuous_projection_net_ctor)
+        return constructor(input_size=input_size, action_spec=spec)
+
+    return alf.nest.map_structure(_create_individually, action_spec)
+
+
+@alf.configurable
+class DisjointPolicyValueNetwork(Network):
+    """A composite network with a policy component and a value component.
+
+    This network capture a category of network as proposed in the Phasic Policy
+    Gradient paper. It consists of two components and 3 heads:
+
+    - Value Component: a single value head that estimates the value function
+
+    - Policy Component: 1 policy head that outputs the action distribution, and
+         1 auxiliary value head that behaves as a secondary value function
+         estimator
+
+    The output of this network is a triplet, corresponding to the 3 heads in the
+    order of (action distribution, value function, auxiliary value function).
+
+    About Architecture:
+
+      The Value Component and the Policy Component may share the same encoding
+      network or have their own encoding network. When the encoding network is
+      shared, it is called the "shared" architecture. If the encoding network is
+      not shared, it is called the "dual" architecture.
+
+      NOTE that in the "shared" architecture, the encoder is detached before
+      connecting to the value head. This means that the value head will have no
+      power to optimize and update the parameters of the encoder under such
+      constraint.
+
+    See https://github.com/HorizonRobotics/alf/issues/965 for a graphical
+    illustration of such two different architectures.
+
+    """
+
+    # TODO(breakds): Add type hints when nest of tensor type is defined
+    def __init__(self,
+                 observation_spec,
+                 action_spec,
+                 encoding_network_ctor=EncodingNetwork,
+                 is_sharing_encoder: bool = False,
+                 kernel_initializer=None,
+                 discrete_projection_net_ctor=CategoricalProjectionNetwork,
+                 continuous_projection_net_ctor=NormalProjectionNetwork,
+                 name='DisjointPolicyValueNetwork'):
+        """The constructor of DisjointPolicyValueNetwork
+
+        Note that there are two projection constructor parameters. They exist
+        because in the case when the action spec is a nest of different types
+        where some of them are discrete and some of them are continuous,
+        corresponding projection networks can be created for the two parties
+        individually and respectively.
+
+        Args:
+
+            observation_spec (nest of TesnorSpec): specifies the shape and type
+                of the input observation.
+
```

Comment:
I think the convention is to remove the empty line between two argument descriptions.

Reply:
Sure, will do.